### DISCUSSION TITLE
Improve Versioning / Diff view

### DISCUSSION THREAD

**Discussion by zubricks** [payloadcms/payload#4197] [Roadmap]
To improve usability + efficiency we’ll continue to make UX improvements to the Versions/Diff views.

**Diff overhaul**
- [X] Done: improved relationship diffing https://github.com/payloadcms/payload/pull/12027
- [X] Done: improved richtext diffing https://github.com/payloadcms/payload/pull/11760
- [X] Done: allow for customization of diff components https://github.com/payloadcms/payload/pull/10815
- [X] Done: more control over versions screens in general https://github.com/payloadcms/payload/pull/10815

**Misc.**
- [ ] e.g. use a different Relations field, replace with a custom one
- [ ] Rich Text is huge, only show the diffs
- [ ] Rich Text JSON block is hard to read through

**Comment by AshtarCodes** [comment]
This is really important for content teams where they are not technical enough to look through a JSON diff. Is there a plan to optionally display a more plain text diff, where changes can be seen more intuitively? Something more similar to how document editors show them? 

Our team is migrating from Confluence, where they render diffs in a more visually intuitive way despite having a layout builder / macros. These are rendered visually in the diff as well. I'm just curious what you all think about that? Do you think there's value for you? How much of a priority is it?

**Comment by vibhor-web-headout** [comment]
Another improvement would be displaying only the diff section instead of the whole page and highlighting.
Pages can be huge, and searching for diffs in between can be painful.

**Comment by tsemachh** [comment]
Any chance the Diff View is broken in payload 3 ?
I only see Status Updated at and Create at

**Comment by PP-Tom** [comment]
https://github.com/payloadcms/payload/discussions/1440 I made a suggestion here that is based on Contentful that I think does it really well. It's in the Roadmap.

**Comment by GermanJablo** [comment]
I think this one can be closed. Is there anything missing here, @AlessioGr?
